cmake/WarpXFunctions.cmake

```
@@ -171,41 +171,36 @@ macro(set_default_build_type default_build_type)
     endif()
 endmacro()
 
-# Set CXX
-# Note: this is a bit legacy and one should use CMake TOOLCHAINS instead.
+# Set compile warnings
 #
-macro(set_cxx_warnings)
+function(set_compile_warnings tgt)
```

We might define similar functions in pyAMReX/ImpactX, so avoiding a collision in superbuilds we can call this:
```suggestion
function(warpx_set_compile_warnings tgt)
```